### PR TITLE
Adds IsRowVersion fluent API

### PR DESCRIPTION
**Pull request by ajcvickers**
Issue #5554

This is pure sugar for ValueGeneratedOnAddOrUpdate and IsConcurrencyToken.

### CODE REVIEW

File: test/Microsoft.EntityFrameworkCore.SqlServer.FunctionalTests/F1SqlServerFixture.cs [hunk 1]
```diff
@@ -60,26 +60,12 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
...
-
-            modelBuilder.Entity<Team>(b =>
-                {
-                    b.Property<byte[]>("Version")
-                        .ValueGeneratedOnAddOrUpdate()
-                        .IsConcurrencyToken();
-                });
+            modelBuilder.Entity<Team>().Property<byte[]>("Version")
+                .ValueGeneratedOnAddOrUpdate()
+                .IsConcurrencyToken();
```
Comment (smitpatel) [inline test/Microsoft.EntityFrameworkCore.SqlServer.FunctionalTests/F1SqlServerFixture.cs]: Is it intentional not to use `IsRowVersion` for this one?

**Comment by smitpatel** [comment]
:shipit:

File: test/Microsoft.EntityFrameworkCore.SqlServer.FunctionalTests/F1SqlServerFixture.cs (hunk 1 above)
Comment (ajcvickers) [inline test/Microsoft.EntityFrameworkCore.SqlServer.FunctionalTests/F1SqlServerFixture.cs]: Yes. Both ways should do the same thing, so this adds coverage for both ways of doing it.

**Comment by roji** [comment]
Are you guys sure this belongs in Microsoft.EFCore rather than SqlServer? The concept of RowVersion really doesn't seems to be very SqlServer-specific...

**Comment by ErikEJ** [comment]
I am assuming that this means: The column type is RowVersion - which is quite SQL Server specific, I believe

**Comment by ajcvickers** [comment]
@roji @ErikEJ The API name is very SQL Server specific, but the behavior is just to do the same as the Timestamp attribute. We discussed this one round in circles many times. In the end we went for the same name and bahevior as EF6 and before, with providers free to implement in a way that makes sense, or to not support this combination of facets.

**Comment by roji** [comment]
Yes sorry, I meant "seems to be very SqlServer-specific" above (edited).

I guess this seems reasonable for backwards compatibility reasons, as long as the database-specific nature of supported concurrency tokens is clearly documented.

**Pull request merged**